orb-ui/src/engine/animations/simple_spinner.rs

```diff
@@ -0,0 +1,218 @@
+                    self.transition_time += dt;
+                    if self.transition_time >= duration {
+                        self.transition = None;
+                        self.transition_background = None;
+                    }
+                    (self.transition_time * PI / 2.0 / duration).sin()
+                }
+                _ => 1.0,
+            };
+
+            // average between background and transition_background
+            // keep intensity of colors in case of a background transition, by resetting factor
+            let background =
+                if let Some(transition_background) = self.transition_background {
+                    let b = transition_background * (1.0 - factor)
+                        + self.background * factor;
+                    factor = 1.0;
+                    b
+                } else {
+                    self.background
+                };
+
+            for (i, led) in frame.iter_mut().enumerate() {
+                if i == led_index {
+                    let c = Argb(
+                        self.color.0,
+                        (self.color.1 as i32
+                            + ((background.1 as i32 - self.color.1 as i32) as f64
+                                * (head_tail_scale))
+                                as i32) as u8,
+                        (self.color.2 as i32
+                            + ((background.2 as i32 - self.color.2 as i32) as f64
+                                * (head_tail_scale))
+                                as i32) as u8,
+                        (self.color.3 as i32
+                            + ((background.3 as i32 - self.color.3 as i32) as f64
+                                * (head_tail_scale))
+                                as i32) as u8,
+                    );
+                    *led = c * factor;
+                } else if i == (led_index + 1) % N || i == (led_index + 2) % N {
+                    *led = self.color * factor;
+                } else if i == (led_index + 3) % N {
+                    let c = Argb(
+                        self.color.0,
+                        (background.1 as i32
+                            + ((self.color.1 as i32 - background.1 as i32) as f64
+                                * head_tail_scale) as i32)
+                            as u8,
+                        (background.2 as i32
+                            + ((self.color.2 as i32 - background.2 as i32) as f64
+                                * head_tail_scale) as i32)
+                            as u8,
+                        (background.3 as i32
+                            + ((self.color.3 as i32 - background.3 as i32) as f64
+                                * head_tail_scale) as i32)
+                            as u8,
+                    );
+                    *led = c * factor;
+                } else {
+                    *led = background * factor;
+                }
+            }
```

Comment:
make it a function called `update_led_frame(&mut self, background, factor)`. Add at least one test case for the function to a unit test.